### PR TITLE
[BUGFIX] Fix typing on `mostly` and `value_set` fields

### PR DESCRIPTION
**Pull request by tyler-hoffman**
## Context
We were previously using `Annotated` incorrectly. The correct way to use it is `Annotated[T, x]`, where `T` is the base case, and `x` is the metadata. We had those type args backward. Pydantic uses the first argument for validation, and schema generation, so schemas were generated as expected, but mypy was upset because e.g. `mostly=1` does not satisfy the type hint that mostly must be an instance of `_Mostly`. So we had to change a bunch of stuff

## About our validation vs schema.
Our validation was fairly lax, so I kept it that way in this PR. e.g. `multiple_of` was not enforced, but was just there for the schema. Pydantic actually has support out of the box for most of what we were doing, including `multiple_of`, but since we didn't enforce this before, I didn't want to start enforcing it now.

## The problem
From experimenting, it looks like constraints on nested annotations are not supported in schema generation. See https://github.com/great-expectations/great_expectations/pull/10577 for how I'd like to have written the solution, and how it fails in CI.

## Implementation details
* We're using the correct ordering in `Annotated` type args
* I'm taking advantage of `pydantic.Field` allowing for arbitrary fields to be passed in AND the `schema_extra` override on the base Expectation class. Basically during `schema_extra`, we pop out a custom field, `schema_overrides`, and write the contents of it to whatever property it was on.

I just copy/pasted what was in the snapshots of our expectation schema json for `value-set`. For `mostly`, I used out of the box functionality where I could, and only passed the `multiple_of` in `schema_overrides`. The change in how these fields are added to the schema results in a bunch of noise in the PR due to ordering changes.

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/expectations/core/schemas/ExpectColumnMostCommonValueToBeInSet.json [hunk 1]
```diff
@@ -81,7 +81,7 @@
         },
         "value_set": {
             "title": "Value Set",
-            "description": "A list of potential values to match.",
```
Comment (tyler-hoffman) [inline great_expectations/expectations/core/schemas/ExpectColumnMostCommonValueToBeInSet.json:84]: This was the only instance where we had a different description for the value set. I think it makes sense to make it the same as the others.

File: great_expectations/expectations/core/schemas/ExpectColumnPairValuesAToBeGreaterThanB.json [hunk 2]
```diff
@@ -80,12 +80,12 @@
         },
         "mostly": {
             "title": "Mostly",
-            "default": 1.0,
```
Comment (tyler-hoffman) [inline great_expectations/expectations/core/schemas/ExpectColumnPairValuesAToBeGreaterThanB.json:83]: Lotta noise on these. It's functionally the same, just a different ordering for `mostly`.

File: great_expectations/expectations/expectation.py [hunk 3]
```diff
@@ -312,6 +312,15 @@ def schema_extra(schema: Dict[str, Any], model: Type[Expectation]) -> None:
...
             }
 
+            # Add extra fields to schema from custom schema_overrides
+            # schema_overrides is not a pydantic concept, but pydantic.Field allows
+            # us to pass through arbitrary fields.
+            for prop in schema["properties"].values():
+                if overrides := prop.pop("schema_overrides", None):
+                    assert isinstance(overrides, dict)
+                    for key, value in overrides.items():
+                        prop[key] = value
```
Comment (joshua-stauffer) [inline great_expectations/expectations/expectation.py]: This is fine, but cleaner might be:
```suggestion
                    prop.update(overrides)
```

**Review by joshua-stauffer — APPROVED** [review]
looks good - thanks for fixing this

File: great_expectations/expectations/expectation.py (hunk 3 above)
Comment (tyler-hoffman) [reply great_expectations/expectations/expectation.py]: Oh yeah, much better.

**Pull request merged**